azavea/cac-tripplanner
# Filter destinations in the map sidebar

## Overview

Implement a dropdown destinations filter for the map sidebar. Continue to use a button bar for filtering on the homepage, and switch between the two as appropriate.

### Demo

![no_events_isochrone](https://user-images.githubusercontent.com/960264/33913251-0b953a86-df67-11e7-9242-060085270a9e.png)


### Notes

The [prototype](https://marvelapp.com/59eh08g/screen/35591677) shows the word 'Filter' at the top of the filter bar, instead of the current selection. That is not typical drop-down behavior, and gives no clear benefit; there will always be a selection, as 'All' is the default. A standard `select` element is in use here.

As visible above, destinations that do not match the filter currently have a lower opacity, as do those outside the isochrone bounds. #937 is to not display locations that do not match the filter at all.

Created #940 to separate place list header and content reloading. Adding the filter bar has changed interaction such that the header shouldn't disappear on list reload; for now, it reloads whenever the list does.


## Testing Instructions

 * `vagrant ssh app`
 * `cd /opt/app/src && npm run gulp-development`
 * Ensure service worker cache is cleared
 * Filtering on home and map pages should behave as expected
 * Navigating between home and map pages should switch between the two filter types


## Checklist
- [x] Gulp tests pass


Closes #933
Closes #937
